altair/vegalite/v5/api.py

```diff
@@ -3629,20 +3643,20 @@ def transformed_data(
 
         return transformed_data(self, row_limit=row_limit, exclude=exclude)
 
-    def __iadd__(self, other: Union[core.LayerSpec, core.UnitSpec]) -> Self:
+    def __iadd__(self, other: Union["LayerChart", Chart]) -> Self:
```

Comment:
All of these seem to have been ignored by type checkers not understanding `.encode`, which like `condition` is now fixed